### DISCUSSION TITLE
My article notes template

### DISCUSSION THREAD

**Discussion by lucasew** [SilentVoid13/Templater#183] [Templates Showcase]
I read some articles via custom automation based on pocket (kindle, the app, telegram, lots of fun stuff) and I've made a template that asks the title and the URL for a quick start.

I just do a Ctrl+M and move it to it's folder.

https://gist.github.com/lucasew/24eac69370f40f947d61d2ba48377811

**Comment by d0verenny** [comment]
Could you please shoot a video of how exactly it works? That'd be so great

**Comment by floris-max** [comment]
<%*
async function handle() {
    const thisNote = app.workspace.getActiveFile();

    // Mostrar un mensaje y esperar entrada válida
    async function prompt(message) {
        let res;
        do {
            res = await tp.system.prompt(message);
        } while (!res);
        return res;
    }

    // Renombrar el archivo actual
    async function renameThisNote(newName) {
        try {
            await app.vault.rename(thisNote, thisNote.path.replace(thisNote.basename, newName));
        } catch (e) {
            new Notice(`Error renaming note: ${e.message || e}`);
            console.error(e);
        }
    }

    // Si el archivo es "Untitled", solicita un nuevo nombre
    if (thisNote.basename.startsWith("Untitled")) {
        const newName = await prompt("File name");
        await renameThisNote(newName);
    }

    // Solicitar URL y modificar contenido
    try {
        const url = await prompt("URL");
        let content = await app.vault.cachedRead(thisNote);
        
        // Reemplazar marcadores en el contenido
        content = content
            .replace("%link%", url)
            .replace("%title%", thisNote.basename);

        await app.vault.modify(thisNote, content);
    } catch (e) {
        new Notice(`Error updating note: ${e.message || e}`);
        console.error(e);
    }
}

// Retrasar la ejecución para trabajar en el archivo recién creado
setTimeout(handle, 200);
%>
